### PR TITLE
Create MultiClusterSecret Controller

### PR DESCRIPTION
**Pull request by desagar**
# Description

Create a controller to handle MultiClusterSecret custom resources (first controller in VZ-2063)

# How has this been tested?

Provide details about how you tested this changed, if necessary
provide instructions to reproduce the testing.

- [x] Tested locally in my own environment
- [x] Successful acceptance test run on CI server
- [ ] Other (specify)

# Checklist 

As the author of this PR, I have:

- [x] Checked my code follows the style guidelines, ran `go fmt`
- [x] Checked that I included or updated copyright and license notices in all files that I altered
- [x] Added or updated godoc for all public functions
- [x] Added or updated unit tests for any new functions I added
- [x] Added or updated integration tests if appropriate
- [ ] Added or updated acceptance tests if appropriate
- [x] Checked that I correctly spelled trademark and product names
- [ ] Used inclusive language and neutral tone
- [x] Not included any sensitive information or hardcoded credentials

Code reviewer, please confirm this PR:

- [ ] Addressed the requirement and meets the acceptance criteria
- [ ] Does not introduce unrelated or spurious changes
- [ ] Does not introduce any unapproved dependency
- [ ] Makes sense and it easy to understand, and/or difficult areas of code are clearly documented so that they can be understood

### CODE REVIEW

File: application-operator/controllers/clusters/multiclustersecret_controller.go [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+}
+
+// +kubebuilder:rbac:groups=clusters.verrazzano.io,resources=multiclustersecrets,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=clusters.verrazzano.io,resources=multiclustersecrets/status,verbs=get;update;patch
+
+// Reconcile reconciles a MultiClusterSecret resource. It fetches the embedded Secret, mutates it
+// based on the MultiClusterSecret, and updates the status of the MultiClusterSecret to reflect the
+// success or failure of the changes to the embedded Secret
+func (r *MultiClusterSecretReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
+	_ = context.Background()
```
Comment (mgianatagh) [inline application-operator/controllers/clusters/multiclustersecret_controller.go:37]: this line can be removed, the assignment is properly taking place on line 41

File: application-operator/controllers/clusters/multiclustersecret_controller.go [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&clustersv1alpha1.MultiClusterSecret{}).
+		Complete(r)
+}
+
+func (r *MultiClusterSecretReconciler) fetchMultiClusterSecret(ctx context.Context, name types.NamespacedName, mcSecretRef *clustersv1alpha1.MultiClusterSecret) error {
+	return r.Get(ctx, name, mcSecretRef)
+}
+
+func (r *MultiClusterSecretReconciler) createOrUpdateSecret(ctx context.Context, mcSecret clustersv1alpha1.MultiClusterSecret, recRes *reconcile.Result) (controllerutil.OperationResult, error) {
```
Comment (mgianatagh) [inline application-operator/controllers/clusters/multiclustersecret_controller.go:114]: The recRes parameter does not appear to be used

File: application-operator/test/integ/multi_cluster_test.go [hunk 3]
```diff
@@ -1,17 +1,22 @@
...
-package integ
+
+package integ_test
 
 import (
 	"fmt"
 	"github.com/onsi/ginkgo"
 	"github.com/onsi/gomega"
 	"github.com/verrazzano/verrazzano/application-operator/test/integ/util"
+	"reflect"
```
Comment (mgianatagh) [inline application-operator/test/integ/multi_cluster_test.go:11]: do you have your IDE set to auto organize your imports?  I noticed a few of them do not appear to be sorted.  The file watcher is "goimports"

File: application-operator/controllers/clusters/multiclustersecret_controller.go (hunk 1 above)
Comment (desagar) [reply application-operator/controllers/clusters/multiclustersecret_controller.go]: done

File: application-operator/controllers/clusters/multiclustersecret_controller.go (hunk 2 above)
Comment (desagar) [reply application-operator/controllers/clusters/multiclustersecret_controller.go]: done

File: application-operator/test/integ/multi_cluster_test.go (hunk 3 above)
Comment (desagar) [reply application-operator/test/integ/multi_cluster_test.go]: I do have it set up but for some reason it doesn't do it automatically. I usually have to tell it to organize them and sometimes I forget. I just did and pushed it for this file.

**Review by mgianatagh — APPROVED** [review]

**Pull request merged**